roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md

```diff
@@ -4,10 +4,324 @@ The coordinator is the central subsystem of the node-side components which parti
+result of that validation (either valid or invalid) will be the node's vote on
+that dispute: Either explicit "invalid" or "valid". The dispute coordinator will
+inform `dispute-distribution` about our vote and `dispute-distribution` will make
+sure that our vote gets distributed to all other validators.
+
+Nothing ever is that easy though. We can not blindly import anything that comes
+along and trigger participation no matter what.
+
+### Spam Considerations
+
+In Polkadot's security model, it is important that attempts to attack the system
+result in a slash of the offenders. Therefore we need to make sure that this
+slash is actually happening. Attackers could try to prevent the slashing from
+taking place, by overwhelming validators with disputes in such a way that no
+single dispute ever concludes, because nodes are busy processing newly incoming
+ones. Other attacks are imaginable as well, like raising disputes for candidates
+that don't exist, just filling up everyone's disk slowly or worse making nodes
+try to participate, which will result in lots of network requests for recovering
+availability.
+
+The last point brings up a significant consideration in general: Disputes are
+about escalation: Every node will suddenly want to check, instead of only a few.
+A single message will trigger the whole network to start significant amount of
+work and will cause lots of network traffic and messages. Hence the
+dispute system is very susceptible to being a brutal amplifier for DoS attacks,
+resulting in DoS attacks to become very easy and cheap, if we are not careful.
+
+One counter measure we are taking is making raising of disputes a costly thing:
+If you raise a dispute, because you claim a candidate is invalid, although it is
+in fact valid - you will get slashed, hence you pay for consuming those
+resources. The issue is: This only works if the dispute concerns a candidate
+that actually exists!
+
+If a node raises a dispute for a candidate that never got included (became
+available) on any chain, then the dispute can never conclude, hence nobody gets
+slashed. It makes sense to point out that this is less bad than it might sound
+at first, as trying to participate in a dispute for a non existing candidate is
+"relatively" cheap. Each node will send out a few hundred tiny request messages
+for availability chunks, which all will end up in a tiny response "NoSuchChunk"
+and then no participation will actually happen as there is nothing to
+participate. Malicious nodes could provide chunks, which would make things more
+costly, but at the full expense of the attackers bandwidth - no amplification
+here. I am bringing that up for completeness only: Triggering a thousand nodes
+to send out a thousand tiny network messages by just sending out a single
+garbage message, is still a significant amplification and is nothing to ignore -
+this could absolutely be used to cause harm!
```

Comment:
AFAIK these 'remote' disputes are still posted on-chain, never conclude because they are by definition only participated in by a superminority, and also lead to a slash when they time out. Although the leak is somewhat slower.

Reply:
This is already assuming [this](https://github.com/paritytech/srlabs_findings/issues/179#user-content-fnref-1-6e95f46a1eeaedce9b8a9461647a0870) proposed alternative handling of potential dispute spam to be in place.